### ISSUE TITLE
check-python-versions starts complaining about mismatch with Python 3.10

### ISSUE THREAD

**Issue by mauritsvanrees** [zopefoundation/meta#123]
I have merged https://github.com/zopefoundation/zope.schema/pull/111 which was a simple typo fix.
The checks were green, but after merge, the lint tests failed when checking Python versions:

```
lint run-test: commands[2] | check-python-versions
setup.py says:                    2.7, 3.5, 3.6, 3.7, 3.8, 3.9, PyPy, PyPy3
- python_requires says:           2.7, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10
tox.ini says:                     2.7, 3.5, 3.6, 3.7, 3.8, 3.9, PyPy, PyPy3
.github/workflows/tests.yml says: 2.7, 3.5, 3.6, 3.7, 3.8, 3.9, PyPy, PyPy3
```

So the `check-python-versions` package has been updated to know about Python 3.10, and now it expects that 3.10 is in the classifiers/tox/gh-actions when you have this in your `setup.py`:

    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',

For this package it will be fixed with PR https://github.com/zopefoundation/zope.schema/pull/109 which was already created beforehand.

But in general, I think most PRs of Zope packages will now fail this test: most will not have 3.10 in classifiers/tox/actions, but `check-python-versions` 0.19.0 will expect it, since [this PR](https://github.com/mgedmin/check-python-versions/pull/30) by @mgedmin, which of course is a good change.

What do we do about this?

- Accept failing `lint` checks for now, assuming that such a failing check does not prevent other checks from running?
- Require `lint` to pass, effectively blocking all PRs until a package has in fact been updated and checked to work with 3.10?
- Disable the `check-python-versions` check for now, or somehow only let it report, without failing?
- Pin `check-python-versions` to 0.18.2, except in packages that already really support Python 3.10?
- Add `<3.10` to `python-requires` on packages that do not yet support it? This makes it harder to test on 3.10.

**Comment by dataflake** [comment]
I personally prefer the simple fix in setup.py `python_requires` to explicitly say `<3.10` for every package that really has not been checked against 3.10. Clearly, we don't want to suggest compatibility just because we had this habit of never disallowing unreleased/untested new Python versions before.

**Comment by jamadden** [comment]
>  I personally prefer the simple fix in setup.py `python_requires` to explicitly say <3.10 for every package that really has not been checked against 3.10.

In general, I think I dislike that philosophy for the pragmatic reason that *most* packages are going to (at least *mostly*) work just fine on the next minor version of Python. After all, most of the time, "adding support for Python 3.x" consists of no code changes at all, just metadata. Exceptions can be considered on a case-by-case basis.

It can take some time for all packages to have their metadata updated and then be released to PyPI, and in the meantime that's preventing users from adopting the new Python version. What if all packages did this? Then when 3.10 is released, no one would be able to install anything, so why bother even upgrading? 

Much worse, though, is that strict `python_requires` like that are preventing users from even *testing* the new Python version with their software if they have a package like that as a dependency. Many users and even distros start testing new Python versions in the pre-release stages, and legitimate bug reports can come out of that. I recently had a bug report about a project I maintain on Python 3.10rc1; the user wouldn't have been able to test my project on that version if it contained "< 3.10", and I wouldn't have had a fix ready to go when 3.10 was officially released (I was already testing 3.10rc1, but 100% test coverage doesn't mean 100% bug free; this was an area my tests didn't cover).

It seems to me that "consenting adults" applies here. Yes, it's untested, and yes there may be bugs, but if it works for you, great! (I've used packages where some modules don't import on some versions of Python, but I wasn't using those modules, so it didn't matter to me.) Hopefully we'll get around to updating, officially supporting, and releasing new versions, but until we do — or if we don't, or at least not in a fashion "fast enough" for users — people at least have the ability to try it.

**Comment by dataflake** [comment]
Looks like @mgedmin is already thinking about a fix to `check-python-versions` which would solve the lint failure at least.
